File: c/estrutura_sequencial/idades.c
#include <stdio.h>
#include <string.h>

void limpar_entrada() 
{
    char c;
    while ((c = getchar()) != '\n' && c != EOF) {}
}

void ler_texto(char *buffer, int length) 
{
    fgets(buffer, length, stdin);
    strtok(buffer, "\n");
} 


int main(void)
{
    int idade1, idade2;
    double media;
    char nome1[50], nome2 [50];
    
    printf("Dados da primeira pessoa:\n");
    printf("Nome: ");
    ler_texto(nome1, 50);
    printf("Idade: ");
    scanf("%i", &idade1);
    
    printf("Dados da segunda pessoa:\n");
    printf("Nome: ");
    limpar_entrada();
    ler_texto(nome2, 50);
    printf("Idade: ");
    scanf("%i", &idade2);
    
    media = (idade1 + idade2) / 2.0;
    
    printf("A idade média de %s e %s é de %.1lf", nome1, nome2, media);
    
} 

File: python/idades.py
print("Dados da primeira pessoa: ")
nome1 = input("Nome: ")
idade1 = float(input("Idade: "))

print("Dados da segunda pessoa: ")
nome2 = input("Nome: ")
idade2 = float(input("Idade: "))

media = (idade1 + idade2) / 2

print(f"A idade media de {nome1} e {nome2} é de {media:.1f} anos")
